appengine/standard/memcache/guestbook/main.py

```diff
@@ -85,9 +85,15 @@ def get_greetings(self, guestbook_name):
         greetings = memcache.get('{}:greetings'.format(guestbook_name))
         if greetings is None:
             greetings = self.render_greetings(guestbook_name)
-            if not memcache.add('{}:greetings'.format(guestbook_name),
-                                greetings, 10):
-                logging.error('Memcache set failed.')
+            try:
+                added = memcache.add('{}:greetings'.format(guestbook_name),
+                                     greetings, 10)
+                if not added:
+                    logging.error('Memcache set failed.')
```

Comment:
Do we know for sure if memcache.add returning False is necessarily a 'key already exists' type of error? The docs say memcache.add returns 'False on error'. (https://cloud.google.com/appengine/docs/python/refdocs/modules/google/appengine/api/memcache#Client.add). Wondering if the False could mean something else in certain circumstances.